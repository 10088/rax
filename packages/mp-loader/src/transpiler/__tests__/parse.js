const { parse } = require('../parser');
const modules = require('../transpileModules');

const transpilerOptions = { modules };

describe('Transpiler parse', () => {
  it('nested', () => {
    const content = `
      <view foo="bar">
        <text>hello</text>
        <text>world</text>
      </view>
    `;
    const ast = parse(content, transpilerOptions);
    expect(ast).toMatchSnapshot();
  });

  it('bind props', () => {
    const content = `
      <view foo="bar {{someVal}} {{otherVal}}">
        <text>hello</text>
        <text>world</text>
      </view>
    `;
    const ast = parse(content, transpilerOptions);
    expect(ast).toMatchSnapshot();
  });

  it('a:for', () => {
    const content = `
      <view a:for="todos"></view>
    `;
    const ast = parse(content, transpilerOptions);
    expect(ast).toMatchSnapshot();
  });

  it('should use a:for with other rules', () => {
    const content = `
      <view a:for="todos" a:key="{{item.id}}" checked="{{item.checked}}"></view>
    `;
    const ast = parse(content, transpilerOptions);
    expect(ast).toMatchSnapshot();
  });
});
