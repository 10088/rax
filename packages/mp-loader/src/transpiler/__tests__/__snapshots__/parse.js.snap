// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transpiler parse a:for 1`] = `
Object {
  "alias": "item",
  "attrsList": Array [],
  "attrsMap": Object {
    "a:for": "todos",
  },
  "children": Array [],
  "for": "todos",
  "iterator1": "index",
  "parent": undefined,
  "plain": false,
  "tag": "view",
  "type": 1,
}
`;

exports[`Transpiler parse bind props 1`] = `
Object {
  "attrs": Array [
    Object {
      "name": "foo",
      "scope": "",
      "value": "(\\"bar \\") + (data['someVal']) + (\\" \\") + (data['otherVal'])",
    },
  ],
  "attrsList": Array [],
  "attrsMap": Object {
    "foo": "bar {{someVal}} {{otherVal}}",
  },
  "children": Array [
    Object {
      "attrsList": Array [],
      "attrsMap": Object {},
      "children": Array [
        Object {
          "text": "hello",
          "type": 3,
        },
      ],
      "parent": [Circular],
      "plain": true,
      "tag": "text",
      "type": 1,
    },
    Object {
      "attrsList": Array [],
      "attrsMap": Object {},
      "children": Array [
        Object {
          "text": "world",
          "type": 3,
        },
      ],
      "parent": [Circular],
      "plain": true,
      "tag": "text",
      "type": 1,
    },
  ],
  "parent": undefined,
  "plain": false,
  "tag": "view",
  "type": 1,
}
`;

exports[`Transpiler parse nested 1`] = `
Object {
  "attrs": Array [
    Object {
      "name": "foo",
      "value": "\\"bar\\"",
    },
  ],
  "attrsList": Array [
    Object {
      "name": "foo",
      "value": "bar",
    },
  ],
  "attrsMap": Object {
    "foo": "bar",
  },
  "children": Array [
    Object {
      "attrsList": Array [],
      "attrsMap": Object {},
      "children": Array [
        Object {
          "text": "hello",
          "type": 3,
        },
      ],
      "parent": [Circular],
      "plain": true,
      "tag": "text",
      "type": 1,
    },
    Object {
      "attrsList": Array [],
      "attrsMap": Object {},
      "children": Array [
        Object {
          "text": "world",
          "type": 3,
        },
      ],
      "parent": [Circular],
      "plain": true,
      "tag": "text",
      "type": 1,
    },
  ],
  "parent": undefined,
  "plain": false,
  "tag": "view",
  "type": 1,
}
`;

exports[`Transpiler parse should use a:for with other rules 1`] = `
Object {
  "alias": "item",
  "attrs": Array [
    Object {
      "name": "checked",
      "scope": "this.data",
      "value": "(item.checked)",
    },
  ],
  "attrsList": Array [],
  "attrsMap": Object {
    "a:for": "todos",
    "a:key": "{{item.id}}",
    "checked": "{{item.checked}}",
  },
  "children": Array [],
  "for": "todos",
  "iterator1": "index",
  "key": "(item.id)",
  "parent": undefined,
  "plain": false,
  "tag": "view",
  "type": 1,
}
`;
